### ISSUE TITLE
Warning: Selenium 3.5.1 breaks java client 5.0.0-BETA9

### ISSUE THREAD

**Issue by welnanick** [appium/java-client#696]
I am using gradle to handle dependencies for my project. In my `build.gradle` I have `testCompile "io.appium:java-client:5.0.0-BETA9"` to compile the appium java client. It seems Selenium 3.5.1 was just released (their website has not been updated as I am writing this), and for some reason gradle is grabbing version 3.5.1 automatically for selenium-api, selenium-remote-driver, and selenium-support. I don't know why this is happening, but it is and it breaks the java client. I had to add
```
compile ('org.seleniumhq.selenium:selenium-api:3.4.0'){force = true}
compile ('org.seleniumhq.selenium:selenium-remote-driver:3.4.0'){force = true}
compile ('org.seleniumhq.selenium:selenium-support:3.4.0'){force = true}
```
to fix my project.

So, if your project just suddenly stopped working for no reason whatsoever, check if these packages were updated, and if so make sure to force version 3.4.0 for them.

**Comment by jason-edstrom** [comment]
I just solved this exact issue on my setup. Same resolution approach, forcing 3.4.0.

**Comment by QAutomatron** [comment]
Faced the same problem. Thank you for a solution. 

I tried a different one, it also helps:
```
configurations.all {
    resolutionStrategy {
        force 'org.seleniumhq.selenium:selenium-support:3.4.0',
                'org.seleniumhq.selenium:selenium-api:3.4.0'
    }
}
```

**Comment by SrinivasanTarget** [comment]
@QAutomatron @welnanick @jason-edstrom Whats the actual problem faced here? What breaks with 3.5.1? Gonna hands-on anyways tomorrow.

**Comment by SrinivasanTarget** [comment]
Changes are here: https://github.com/SeleniumHQ/selenium/commit/323132d747fc5a64e57efd34f392f752073da503

**Comment by welnanick** [comment]
It seems annotations (or possible all FindBy operations given the changes in selenium) break. When testing android, elements populated using `@AndroidFindBy` stay null, and when testing iOS elements populated using `@iOSFindBy` cause `org.openqa.selenium.WebDriverException`

Unfortunately I am not in a position to investigate further, nor confirm whether it is just presenting itself with gradle or not.

**Comment by SrinivasanTarget** [comment]
Cool thanks. will investigate further.

**Comment by SrinivasanTarget** [comment]
@TikhomirovSergey ping

**Comment by Brian-Watson** [comment]
I'm seeing this for my Android tests after the Selenium 3.5.1 upgrade...

`org.openqa.selenium.remote.RemoteWebElement cannot be cast to io.appium.java_client.MobileElement
`
Where my code is attempting this...

`MobileElement foundElement = (MobileElement) wait.until(ExpectedConditions.presenceOfElementLocated(this.getLocator()));
`

**Comment by SrinivasanTarget** [comment]
https://github.com/appium/java-client/pull/698

**Comment by TikhomirovSergey** [comment]
The fix is here #699. It will be published on the nearest (26.08.2017 - 27.08.2017)  weekend.
